internals/webhooks/subnamespace_webhook.go

```diff
@@ -161,9 +161,38 @@ func (a *SubNamespaceAnnotator) Handle(ctx context.Context, req admission.Reques
 			return admission.Allowed(allowMessageValidateQuotaObj)
 		}
 
-		if err := ValidateUpdateSnsRequest(parentQuotaObj, sns, oldSns, myQuotaObj); err != nil {
-			return admission.Denied(err.Error())
+		isRp, _ := sns.Object.GetLabels()[danav1.ResourcePool]
+		isUpperRp, _ := sns.Object.GetAnnotations()[danav1.IsUpperRp]
+		// if subnamespace or upper resourcepool - resources validation
+		// reduce desired resources from parent and check validity
+		if isRp != "true" || isUpperRp == danav1.True {
+			if err := ValidateUpdateSnsRequest(parentQuotaObj, sns, oldSns, myQuotaObj); err != nil {
+				return admission.Denied(err.Error())
+			}
```

Comment:
Should probably change the comment here to something a bit more explicit and clearer. Something along the lines of:

"Only check the validity of the request if the subnamespace is not a ResourcePool OR the subnamespace is the upper ResourcePool; this is because only in that case the subnamespace would have a RQ or CRQ attached to it. Otherwise, the subnamespace is part of a ResourcePool (and does not have a RQ/CRQ attached to it) and hence this check is unneeded."